src/NLog/LoggerImpl.cs

```diff
@@ -55,19 +55,35 @@ internal static void Write([NotNull] Type loggerType, [NotNull] TargetWithFilter
         {
 #if !NETSTANDARD1_0 || NETSTANDARD1_5
             StackTraceUsage stu = targetsForLevel.GetStackTraceUsage();
-            if (stu != StackTraceUsage.None && !logEvent.HasStackTrace)
+            if (stu != StackTraceUsage.None)
             {
+                bool attemptCallSiteOptimization = targetsForLevel.TryCallSiteClassNameOptimization(stu, logEvent);
+                if (attemptCallSiteOptimization && targetsForLevel.TryLookupCallSiteClassName(logEvent, out string callSiteClassName))
+                {
+                    logEvent.CallSiteInformation.CallerClassName = callSiteClassName;
```

Comment:
shouldn't we check ` logEvent.CallSiteInformation` for null?

Reply:
or should we use `GetCallSiteInformationInternal()`?

Reply:
There is no need for null check as it needs the callerinfo-details to make the lookup (details equals null means no lookup and no assignment of class-name)